docs/library/datadisplay/moment.md

```diff
@@ -0,0 +1,132 @@
+to_now_example = "rx.moment(MomentState.date_now, to_now=True)"
+from_now_during_example = "rx.moment(MomentState.date_now, from_now_during=100000)  # after 100 seconds, date will display normally"
+
+
+format_example_1 = 'rx.moment(MomentState.date_now, format="YYYY-MM-DD")'
+format_example_2 = 'rx.moment(MomentState.date_now, format="HH:mm:SS")'
+
+
+add_example = """rx.vstack(
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(years=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(quarters=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(months=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(months=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(months=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(weeks=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(days=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(hours=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(minutes=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, add=rx.MomentDelta(seconds=2), format="YYYY-MM-DDTHH:mm:SS"),
+)
+"""
+
+subtract_example = """rx.vstack(
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(years=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(quarters=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(months=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(months=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(months=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(weeks=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(days=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(hours=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(minutes=2), format="YYYY-MM-DDTHH:mm:SS"),
+    rx.moment(MomentState.date_now, subtract=rx.MomentDelta(seconds=2), format="YYYY-MM-DDTHH:mm:SS"),
+)
+"""
+
+timezone_example1 = """rx.vstack(
+    rx.moment(MomentState.date_now, tz="America/Los_Angeles"),
+    rx.moment(MomentState.date_now, tz="Europe/Paris"),
+    rx.moment(MomentState.date_now, tz="Asia/Tokyo"),
+)
+"""
+```
+
+## Examples
+
+Using the same date as a value, we will display it in a few different way using `rx.moment`
+
+```python
+class MomentState(State):
+    date_now: datetime = datetime.now()
+```
+
+### Display the date as-is:
+
+```python eval
+docdemo(
+    as_is_example,
```

Comment:
personally i would `python exec` the examples right above where they are used for better locality and editing.

one of my biggest annoyances with the old docs was having to scroll up to the top to see what `example3` meant; with flexdown we can put the example code and the rendering next to each other